extensions/azure/assetindex-cosmos/src/test/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndexTest.java

```diff
@@ -87,114 +80,87 @@ void inputValidation() {
     void findById() {
         String id = "id-test";
         AssetDocument document = createDocument(id);
-        expect(api.queryItemById(eq(id))).andReturn(document);
-
-        replay(api);
-
-        CosmosAssetIndex assetIndex = new CosmosAssetIndex(api, TEST_PARTITION_KEY, typeManager, retryPolicy);
+        when(api.queryItemById(eq(id))).thenReturn(document);
 
         Asset actualAsset = assetIndex.findById(id);
-        assertThat(actualAsset.getProperties()).isEqualTo(document.getWrappedAsset().getProperties());
 
-        verify(api);
+        assertThat(actualAsset.getProperties()).isEqualTo(document.getWrappedAsset().getProperties());
+        verify(api).queryItemById(eq(id));
```

Comment:
this would be a situation where `verifyNoMoreInteractions` would actually contribute to the test strictness, since `api` was a strictmock before. same for all other testmethods in this class.